cirq/ops/fsim_gate.py

```diff
@@ -301,6 +301,15 @@ def rz_angles_after(
         a1 = (-self.gamma - self.zeta + self.chi) / 2.0
         return a0, a1
 
+    def qubit_index_to_equivalence_group_key(self, index: int) -> int:
+        """Returns a key that differs between non-interchangeable qubits."""
+        # PhasedFSimGate is symmetric with respect to qubit swap
+        # if and only if zeta and chi are integer multiples of pi.
+        r = (-np.pi, 0.0, np.pi)
+        if self.zeta in r and self.chi in r:
```

Comment:
It would be very nice to use isclose for this check but I guess there is a problem how to pass tolerance here. (I'm fine with the exact comparison as well.)

Reply:
Yeah, exact comparisons of floating point numbers make me unhappy, too. Unfortunately, this is a recurring issue in cirq and in past discussions we settled on using such comparisons in some cases (e.g. when algebraic properties are at stake as here) on the premise that we're interested in the case when a user provides a special value such as pi exactly (as opposed to the value resulting from a floating point computation).

This is in some sense more appropriate because for any other value the gate is not strictly speaking symmetric with respect to qubit exchange (even though it may be approximately symmetric).

That said, I'm open to doing it either way if your use-case suggests we should do something else.

Reply:
Thanks for explanation, this is perfectly fine withe me - I personally don't intend to use the interchangeability property in anything that I can think of right now.